Build issue:  immintrin.h: included on non-x86 arches

# Summary
There is a compilation issue on non-x86 arches where immintrin.h is included despite non been available on aarch64 and ppc64le compiler.
The following message can be seen:
  `oneDNN-2.3.2/src/gpu/jit/ngen/ngen_utils.hpp:22:10: fatal error: immintrin.h: No such file or directory`

# Version
2.3.2

# Environment
* CPU: aarch64 & ppc64le
* OS version: Fedora 35
* Compiler version: gcc-11
* CMake version : 2.21
* CMake output log

# Steps to reproduce
Build with   -DDNNL_GPU_RUNTIME=OCL -DDNNL_ARCH_OPT_FLAGS="" 

# Observed behavior
  `oneDNN-2.3.2/src/gpu/jit/ngen/ngen_utils.hpp:22:10: fatal error: immintrin.h: No such file or directory`

# Expected behavior
Build should pass on theses arches.

I'm currently testing few cases:
- if the issue can be workaround by any cmake build option
- also if using #ifdef __INTEL_COMPILER guard before the header (as done in others cases) will help...
Will report.

Side note: I'm the fedora package maintainer of onednn.

--- a/src/gpu/jit/ngen/ngen_utils.hpp
+++ b/src/gpu/jit/ngen/ngen_utils.hpp
@@ -19,8 +19,6 @@
 
 #include <cstdint>
 
-#include <immintrin.h>
-
 #ifdef _MSC_VER
 #include <intrin.h>
 #endif

